app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -121,14 +120,6 @@ async function addEthereumChainHandler(
     );
   }
 
-  if (CHAIN_ID_TO_NETWORK_ID_MAP[_chainId]) {
-    return end(
-      ethErrors.rpc.invalidParams({
-        message: `May not specify default MetaMask chain.`,
-      }),
-    );
-  }
-
```

Comment:
I am not sure, id there any reason this check was put in place initially ?

Reply:
I could not find any documentation or information in commit history as to why that check is there. The same functionality is permitted through manual adding of RPC networks. In both flows, warnings are presented with regards to existing chain id networks.